rules/library/xcconfig.bzl

```
@@ -338,3 +325,30 @@ def copts_by_build_setting_with_defaults(xcconfig = {}, fetch_default_xcconfig =
         ibtool_copts = select(ibtool_copts_by_build_setting),
         linkopts = select(linkopts_by_build_setting),
     )
+
+def _merge_xcconfigs(*xcconfigs):
+    """Merges a list of xcconfigs into a single dictionary
+
+    Uses some heuristics to merge xcconfigs in a way that is compatible with Xcode's behavior:
+
+    - If a key is present in multiple xcconfigs
+        - If the value is a list, the values are concatenated
+        - If the value is a string, the last xcconfig value is used
+
+    Args:
+        *xcconfigs: A list of dictionaries of Xcode build settings
+    Returns:
+        A dictionary of Xcode build settings
+    """
+    merged_xcconfig = {}
+    for xcconfig in xcconfigs:
+        for (key, value) in dict(xcconfig).items():
+            if key in merged_xcconfig:
+                if types.is_list(value):
+                    merged_xcconfig[key] = merged_xcconfig[key] + value
+                else:
+                    merged_xcconfig[key] = value
```

What about something like:
```suggestion
            if types.is_list(merged_xcconfig.get(key, None)) and types.is_list(value):
                merged_xcconfig[key] = merged_xcconfig[key] + value
```
I _think_ we should check that both sides are lists (not exactly sure what happens if you misconfigure something with mixed types), and this should also eliminate the need for the second `else`.

---

That's better I'll update. I don't believe we fail if types are misconfigured, but the macros assume lists for some of these xcconfigs. 

We could add failing as a follow up, CocoaPods source has a list of all the list type xcconfig keys.